modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/ClayCriteriaBuilder.es.js

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ClayCriteriaGroup from './ClayCriteriaGroup.es';
+import ClayButton from './ClayButton.es';
+
+class ClayCriteriaBuilder extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			editing: false,
+			initialCriteria: this.props.criteria
+		};
+	}
+
+	render() {
+		const {conjunctions, criteria, operators, properties} = this.props;
+
+		const {editing} = this.state;
+
+		return (
+			<div className="criteria-builder">
+				<div className="criteria-builder-toolbar">
+					<ClayButton
+						label={Liferay.Language.get('edit')}
+						onClick={this._handleToggleEdit}
+					/>
+				</div>
+
+				{this._isCriteriaEmpty() ? (
+					<ClayCriteriaGroup
+						conjunctions={conjunctions}
+						criteria={criteria}
+						criteriaTypes={ClayCriteriaBuilder._buildCriteriaTypes(
+							operators
+						)}
+						editing={editing}
+						onChange={this._updateCriteria}
+						operators={operators}
+						properties={properties}
+						root
+					/>
+				) : (
+					<div
+						className="empty-state"
+						onClick={this._handleNewCriteria}
+					>
+						{Liferay.Language.get('click-to-start-editing')}
+					</div>
+				)}
+			</div>
+		);
+	}
+
+	static _buildCriteriaTypes(operators) {
+		return operators.reduce(
+			(criteriaTypes, {supportedTypes}) => {
+				supportedTypes.forEach(
+					type => {
+						if (!criteriaTypes[type]) {
+							criteriaTypes[type] = operators.filter(
+								operator =>
+									operator.supportedTypes.includes(type)
+							);
+						}
+					}
+				);
+
+				return criteriaTypes;
+			},
+			new Map()
+		);
+	}
+
+	_cleanCriteria(criterion) {
+		const test = criterion
+			.filter(
+				({items}) => {
+					return items ? items.length : true;
```

Comment:
you can use an implicit arrow return here

Reply:
This had to be written this way because the eslint error `no-confusing-arrow` gets thrown. https://eslint.org/docs/rules/no-confusing-arrow